src/TodoContext.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from './types';
+
+type ContextProps = {
+  todo: Todo[],
+  setTodo: (todo: Todo[]) => void,
+  finishedTodos: Todo[],
+  setFinishedTodos: (todo: Todo[]) => void,
+  activeTodos: Todo[],
+  setActiveTodos: (todo: Todo[]) => void,
+  todoses: Todo[],
+  setVisibleTodoses: (todo: Todo[]) => void,
+};
+
+export const TodoContext = React.createContext<ContextProps>({
+  todo: [],
+  setTodo: () => { },
+  finishedTodos: [],
+  setFinishedTodos: () => { },
+  activeTodos: [],
+  setActiveTodos: () => { },
+  todoses: [],
+  setVisibleTodoses: () => { },
+});
+
+export const TodoProvider: React.FC = ({ children }) => {
+  const [todo, setTodo] = useState<Todo[]>(
```

Comment:
when you name an array just todo in the singular it can be confusing for other developers
an object -> todo
array of objects -> todos (plural)

```suggestion
  const [todos, setTodos] = useState<Todo[]>(
```